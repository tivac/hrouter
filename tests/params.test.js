"use strict";

const hrouter = require("../src/router.js");

describe("hrouter", () => {
    beforeEach(() => {
        hrouter._routes = Object.create(null);
    });

    describe("params", () => {
        it("should register parameterized routes", () => {
            hrouter("/:fooga", jest.fn());
    
            expect(hrouter._routes).toMatchSnapshot();
        });

        it("should register nested parameterized routes", () => {
            hrouter("/:fooga/:booga", jest.fn());

            expect(hrouter._routes).toMatchSnapshot();
        });
        
        it("should register parameterized routes w/ multiple callbacks", () => {
            hrouter("/:fooga", jest.fn(), jest.fn(), jest.fn());
    
            expect(hrouter._routes).toMatchSnapshot();
        });
        
        it("should register nested parameterized routes w/ multiple callbacks", () => {
            hrouter("/:fooga", jest.fn(), jest.fn(), jest.fn());
            hrouter("/:fooga/:booga", jest.fn(), jest.fn());
    
            expect(hrouter._routes).toMatchSnapshot();
        });
    
        it("should match simple parameterized routes", () => {
            const fn = jest.fn((ctx, next) => next());
    
            hrouter("/:fooga", fn);
    
            hrouter.go("/fooga");
    
            expect(fn.mock.calls).toMatchSnapshot();
        });
    
        it("should match nested parameterized routes", () => {
            const fn = jest.fn((ctx, next) => next());
    
            hrouter("/:fooga", fn);
            hrouter("/:fooga/:booga", fn);
    
            hrouter.go("/fooga/booga");
    
            expect(fn.mock.calls).toMatchSnapshot();
        });
    });
});
