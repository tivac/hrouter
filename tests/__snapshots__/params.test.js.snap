// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hrouter params should match nested parameterized routes 1`] = `
Array [
  Array [
    Object {
      "params": Object {
        "booga": "booga",
        "fooga": "fooga",
      },
      "url": "/fooga/booga",
    },
    [Function],
  ],
  Array [
    Object {
      "params": Object {
        "booga": "booga",
        "fooga": "fooga",
      },
      "url": "/fooga/booga",
    },
    [Function],
  ],
]
`;

exports[`hrouter params should match simple parameterized routes 1`] = `
Array [
  Array [
    Object {
      "params": Object {
        "fooga": "fooga",
      },
      "url": "/fooga",
    },
    [Function],
  ],
]
`;

exports[`hrouter params should register nested parameterized routes 1`] = `
Object {
  "__param__": Object {
    "__param__": Object {
      "fns": Array [
        [MockFunction],
      ],
      "param": "booga",
    },
    "param": "fooga",
  },
}
`;

exports[`hrouter params should register nested parameterized routes w/ multiple callbacks 1`] = `
Object {
  "__param__": Object {
    "__param__": Object {
      "fns": Array [
        [MockFunction],
        [MockFunction],
      ],
      "param": "booga",
    },
    "fns": Array [
      [MockFunction],
      [MockFunction],
      [MockFunction],
    ],
    "param": "fooga",
  },
}
`;

exports[`hrouter params should register parameterized routes 1`] = `
Object {
  "__param__": Object {
    "fns": Array [
      [MockFunction],
    ],
    "param": "fooga",
  },
}
`;

exports[`hrouter params should register parameterized routes w/ multiple callbacks 1`] = `
Object {
  "__param__": Object {
    "fns": Array [
      [MockFunction],
      [MockFunction],
      [MockFunction],
    ],
    "param": "fooga",
  },
}
`;
